Document bugfinder architecture (#130)
Gamify-IT/docs: dev-manuals/architecture/bugfinder/README.md

# Architecture of `Bugfinder`

## Purpose

> The frontend should be an easy to use web application for finding bugs in code snippets and correcting them. At the moment are just example codes provieded in a config file, for the future (currently in progress) a backend provides these code snippets which will be loaded from the backend.

Reviewer: ```suggestion
Being a small minigame to lean programming by finding bugs in code snippets and correcting them.

## General code structure

> ```md
> This backend has the following package structure
> - `assets`: Contains assets like pictures
> - `dummy`: Dummy data like code snippets and their solution
> - `models`: Contains the models
> - `service`: Contains the game logic
> - `store`: Contains the vue store (not used)
> - `views`: Contains the views
> ```

Reviewer: ```suggestion
This repository has the following package structure:

- `assets`: Contains assets like pictures
- `components`: Contains re-usable Vue components
- `dummy`: Dummy data like code snippets and their solution
- `models`: Contains the data models
- `service`: Contains the game logic
- `store`: Interacts with the vue state store vuex (currently unused)
- `views`: Contains the complete views displayed by this single-page application
```

## Starting the service

> See the [README](https://github.com/Gamify-IT/bugfinder/blob/main/README.md)

Reviewer: ```suggestion
See the [README](https://github.com/Gamify-IT/bugfinder#readme).
```

## What to look at

- everything

## What to ignore

- nothing

## Known Design Flaws

> - no knowing issues

Reviewer: ```suggestion
- no known issues
```

## Known Bugs

An up-to-date list of open bugs can be found at <https://github.com/orgs/Gamify-IT/projects/6/views/11> by searching for `bugfinder`.

> At the moment no knowing issues.

Reviewer: ```suggestion
```
